setup/PowerShellSetup.Test/MsiLayoutTests.cs

```diff
@@ -0,0 +1,270 @@
+    /// These set of tests are especially for scenario like:
+    ///     1) Layout errors, things getting copied to the wrong location
+    ///     2) Checking if files in the MSI including the MSI are signed
+    ///     3) Checking if files included in MSI uses right signing alogirthm
+    /// 
+    /// </summary>
+    public class MsiLayoutTests: MsiTestBase
+    {
+        const string MSI_NAME = "AzurePowerShell.msi";
+        const string MSI_EXTRACT_DIR_NAME = "msiContents";
+        string _procOutput;
+        string _procErr;
+
+        public MsiLayoutTests(ITestOutputHelper testOutput) : base(testOutput)
+        {
+            _procOutput = string.Empty;
+            _procErr = string.Empty;
+        }
+
+        [Fact]
+        [Trait("AcceptanceType", "CheckIn")]
+        public void VerifyMsiExecExists()
+        {
+            ProcessStartInfo psi = GetInitializedPSI();
+            psi.FileName = "Msiexec.exe";
+            psi.Arguments = "/qn /x:1234";
+
+            this.ExecuteShellCmd(psi, out _procOutput, out _procErr);
+            Assert.NotEmpty(_procOutput);
+        }
+        
+        [Fact]
+        [Trait("AcceptanceType", "CheckIn")]
+        public void VerifyNoJavaScriptFiles()
+        {
+            string procErr = string.Empty;
+            string msiContentsDirPath = ExtractMsiContents(out procErr);
+            IEnumerable<string> msiFiles = Directory.EnumerateFiles(msiContentsDirPath, "*.js", SearchOption.AllDirectories);
+            TestLog.WriteLine("Expecting no *.js files in MSI");            
+            foreach (string unsigFile in msiFiles)
+            {
+                TestLog.WriteLine(unsigFile);
+            }
+            Assert.Equal(0, msiFiles.Count<string>());
+        }
+
+        [Fact]
+        [Trait("AcceptanceType", "CheckIn")]
+        public void VerifyNoJsonFiles()
+        {
+            string procErr = string.Empty;
+            string msiContentsDirPath = ExtractMsiContents(out procErr);
+            IEnumerable<string> msiFiles = Directory.EnumerateFiles(msiContentsDirPath, "*.json", SearchOption.AllDirectories);
+            TestLog.WriteLine("Expecting no *.json files in MSI");
+            foreach (string unsigFile in msiFiles)
+            {
+                TestLog.WriteLine(unsigFile);
+            }
+            Assert.Equal(0, msiFiles.Count<string>());
+        }
+        
+        [Fact]
+        [Trait("SignedBuild", "BVT")]
+        public void VerifyFilesAreSigned()
+        {
+            string SHA1 = "sha1RSA";
+            string SHA2 = "sha256RSA";
+
+            List<string> expectedSignatureAlgos = new List<string>() { SHA1, SHA2 };
+            string msiContentsDir = this.ExtractMsiContents(out _procErr);
+            Assert.True(Directory.Exists(msiContentsDir));
+            Assert.True(string.IsNullOrEmpty(_procErr));
+
+            IEnumerable<string> msiFiles = Directory.EnumerateFiles(msiContentsDir, "*", SearchOption.AllDirectories);
```

Comment:
@markcowl the primary reason to do this is to find out whatever is laid down on user machine from the msi is properly signed. We can change our build system and might include unsigned files, so this will insure we have all properly signed files.